game-core/src/main/java/games/strategy/engine/data/export/GameDataExporter.java

```diff
@@ -267,409 +211,348 @@ private void relationshipInitialize(final GameData data) {
-          .append("\" value=\"")
-          .append(current.getSecond())
-          .append("\"/>\n");
-      if (current.getFirst().equals("occupiedTerrOf")) {
-        alreadyHasOccupiedTerrOf = true;
-      }
-    }
-    // add occupiedTerrOf until we fix engine to only use originalOwner
-    if (!alreadyHasOccupiedTerrOf && attachment instanceof TerritoryAttachment) {
-      final TerritoryAttachment ta = (TerritoryAttachment) attachment;
-      if (ta.getOriginalOwner() != null) {
-        sb.append("            <option name=\"occupiedTerrOf\" value=\"")
-            .append(ta.getOriginalOwner().getName())
-            .append("\"/>\n");
-      }
-    }
-    return sb.toString();
-  }
 
-  private void printAttachments(
-      final Tuple<IAttachment, List<Tuple<String, String>>> attachmentPlusValues) {
-    final IAttachment attachment = attachmentPlusValues.getFirst();
-    try {
-      // TODO: none of the attachment exporter classes have been updated since TripleA version
-      // 1.3.2.2
-      final String attachmentOptions;
-      attachmentOptions =
-          printAttachmentOptionsBasedOnOriginalXml(attachmentPlusValues.getSecond(), attachment);
-      final NamedAttachable attachTo = (NamedAttachable) attachment.getAttachedTo();
-      // TODO: keep this list updated
-      String type = "";
-      if (attachTo.getClass().equals(GamePlayer.class)) {
-        type = "player";
-      }
-      if (attachTo.getClass().equals(UnitType.class)) {
-        type = "unitType";
-      }
-      if (attachTo.getClass().equals(Territory.class)) {
-        type = "territory";
-      }
-      if (attachTo.getClass().equals(TerritoryEffect.class)) {
-        type = "territoryEffect";
-      }
-      if (attachTo.getClass().equals(Resource.class)) {
-        type = "resource";
-      }
-      if (attachTo.getClass().equals(RelationshipType.class)) {
-        type = "relationship";
-      }
-      if (TechAdvance.class.isAssignableFrom(attachTo.getClass())) {
-        type = "technology";
-      }
-      if (type.isEmpty()) {
-        throw new AttachmentExportException(
-            "no attachmentType known for " + attachTo.getClass().getCanonicalName());
-      }
-      if (attachmentOptions.length() > 0) {
-        xmlfile
-            .append("        <attachment name=\"")
-            .append(attachment.getName())
-            .append("\" attachTo=\"")
-            .append(attachTo.getName())
-            .append("\" javaClass=\"")
-            .append(attachment.getClass().getCanonicalName())
-            .append("\" type=\"")
-            .append(type)
-            .append("\">\n");
-        xmlfile.append(attachmentOptions);
-        xmlfile.append("        </attachment>\n");
+          unitPlacements.add(
+              Initialize.UnitInitialize.UnitPlacement.builder()
+                  .owner(
+                      player == null || player.getName().equals(Constants.PLAYER_NAME_NEUTRAL)
+                          ? null
```

Comment:
Codacy found an issue: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=6193173)